blobfile/_context.py

```diff
@@ -115,7 +116,7 @@ def copy(
                 copy_fn = _parallel_download
 
             if _is_local_path(src) and _is_azure_path(dst):
-                copy_fn = azure.parallel_upload
+                copy_fn = partial(azure.parallel_upload, version=version)
```

Comment:
not sure if this is the cleanest way of doing this, but will cause fail-fast trying to pass versions into gcp or local fns.

Reply:
I don't follow, how does this cause a failure when using GCP?

Reply:
I think the remote copy case was missed as well, going to fix some things up on the other PR, might have to merge tomorrow.